lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -162,6 +162,41 @@ static Value getScalarIntValue(Value input, Location loc,
   return nullptr;
 }
 
+static Value getScalarFloatValue(Value input, Location loc,
+                                 PatternRewriter &rewriter) {
+  auto inputType = input.getType();
+  if (inputType.isa<Torch::FloatType>()) {
+    return input;
+  }
+
+  auto inputTensorType = inputType.dyn_cast<BaseTensorType>();
+  if (!inputTensorType)
+    return nullptr;
+
+  Type inputDtype = inputTensorType.getOptionalDtype();
+  if (!inputDtype ||
+      (!inputDtype.isF16() && !inputDtype.isF32() && !inputDtype.isF64()))
+    return nullptr;
+
+  std::optional<unsigned> inputRank = getTensorRank(input);
+  if (!inputRank || *inputRank != 0)
+    return nullptr;
+
+  if (auto valueTensorLiteralOp = input.getDefiningOp<ValueTensorLiteralOp>()) {
+    auto val = valueTensorLiteralOp.getValue()
+                   .cast<DenseElementsAttr>()
+                   .getSplatValue<float>();
```

Comment:
I think this should be `double`, since the input type can be an `f64`

Reply:
I think `getSplatValue<APFloat>().getValueAsDouble()` is more general for different float type.